Cargo.toml

```diff
@@ -24,8 +24,11 @@ k8s-openapi = { version = "0.13.0", default-features = false, features = ["v1_22
 kube = "0.64.0"
 kubewarden-policy-sdk = "0.2.2"
 lazy_static = "1.4.0"
+oci-distribution = { git = "https://github.com/kubewarden/krustlet", branch = "oci-distribution-fix-push-to-ghcr"}
+policy-fetcher = { git = "https://github.com/kubewarden/policy-fetcher", tag = "v0.2.0" }
```

Comment:
I am only a bit concerned about policy-evaluator depending on policy-fetcher. This is not a blocker, but I would prefer that they don't depend on each other, maybe by splitting common behaviors in other crate that both depend on.

Reply:
The main issue we have is: we need to interact with a container registry, to do that we need a docker configuration file and a file defining the sources (what is insecure, what is using a self-signed certificate,...).

These data types (`DockerConfig`, `Sources`) are defined inside of policy-fetcher. This crate has also all the conversion functions that can turn these data into something that `oci-distribution` can understand.

I have to admit my head :boom: every time I have to do all these conversions... And I had to do basically the same thing with the sigstore-rs crate, in order to not expose the mutable (as in they could change over the time) `oci-distribution` types into my public API. It's just a gigantic mess...

Sorry about the rant, I really don't have a better idea about how to tackle that

Reply:
It's fine, in this case I prefer to have it working, and see if we can refactor later.